### PR TITLE
feat : Add positive and negative trait impl tests for SIMD types #130

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
Hey, this is my first open-source contribution. I have added the tests as instructed to the best of my knowledge. Please do suggest changes to it if you find something wrong.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -4835,5 +4835,58 @@ mod tests {
...
         assert_impls!([NotZerocopy; 1]: !FromZeroes, !FromBytes, !AsBytes, !Unaligned);
+
+        #[cfg(feature = "simd")] 
+        {
+            macro_rules! test_simd_arch_mod {
+                ($arch:ident, $($typ:ident),*) => {
+                    {
+                        use core::arch::$arch::{$($typ),*};
+                        use crate::*;
+                        $( assert_impls!($ty: KnownLayout, FromZeroes, FromBytes, AsBytes, !Unaligned); )*
```
Comment (jswrenn) [inline src/lib.rs:4846]:
```suggestion
                        $( assert_impls!($typ: KnownLayout, FromZeroes, FromBytes, AsBytes, !Unaligned); )*
```

File: src/lib.rs
Comment (jswrenn) [inline src/lib.rs:1]: Can you run `cargo fmt`? This PR is currently failing the CI formatting check: https://github.com/google/zerocopy/actions/runs/6478364799/job/17590006383?pr=487

**Review by jswrenn — CHANGES_REQUESTED** [review]
Thank you! This looks good. A few minor changes are required to get this passing CI, and you'll also need to sign the Google CLA.

File: src/lib.rs (hunk 1 above)
Comment (Sh0g0-1758) [reply src/lib.rs]: Fixed this issue.

File: src/lib.rs
Comment (Sh0g0-1758) [reply src/lib.rs:1]: ran the command and pushed the formatted code.

**Comment by Sh0g0-1758** [comment]
@jswrenn , I have made the changes and all the CLI checks are passing. Can you please review it once.

**Review by joshlf — APPROVED** [review]

**Comment by Sh0g0-1758** [comment]
@joshlf, it seems that the merge did not happen.

**Comment by joshlf** [comment]
> @joshlf, it seems that the merge did not happen.

This is failing due to an unrelated error: https://github.com/rust-lang/miri/issues/3125. I'll merge manually.

**Pull request merged**